### PR TITLE
metric learning tutorial

### PR DESCRIPTION
**Pull request by matpalm**
A simple metric learning example using CIFAR-10 to train an embedding model.

( exported from a colab if that works? )

### CODE REVIEW

File: examples/tutorial/metric-learning.ipynb [hunk 1]
```diff
@@ -0,0 +1,668 @@
+{
```
Comment (kihyuks) [inline examples/tutorial/metric-learning.ipynb:1]: <p>remove double quotation mark between learning and tutorial</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/google/objax/pull/24/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Comment (kihyuks) [inline examples/tutorial/metric-learning.ipynb:1]: <p>Would you please add a following citation on the sampling technique introduced here: N-pair sampling (<a href="https://papers.nips.cc/paper/6200-improved-deep-metric-learning-with-multi-class-n-pair-loss-objective" target="_blank">https://papers.nips.cc/paper/6200-improved-deep-metric-learning-with-multi-class-n-pair-loss-objective</a>)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/google/objax/pull/24/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='9'/>

Comment (kihyuks) [inline examples/tutorial/metric-learning.ipynb:1]: <p>remove double quotation between learning and tutorial</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/google/objax/pull/24/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='15'/>

Comment (kihyuks) [inline examples/tutorial/metric-learning.ipynb:1]: <p>near neighbours -&gt; either nearest neighbours or neighbours</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/google/objax/pull/24/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='21'/>

Comment (kihyuks) [inline examples/tutorial/metric-learning.ipynb:1]: <p>near neighbours -&gt; nearest neighbours or neighbours</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/google/objax/pull/24/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='25'/>

Comment (matpalm) [reply examples/tutorial/metric-learning.ipynb:1]: changed all references in text and code to "nearest neighbours"

**Comment by matpalm** [comment]
PTAL

**Comment by kihyuks** [comment]
Thanks for an awesome tutorial on metric learning!!

I am wondering whether you would be also interested in writing a similar tutorial on [self-supervised contrastive learning](https://arxiv.org/abs/2002.05709) -- I can see that the change would be: 
1. adding data augmentation 
2. sampling a pair of images from a single image instead of images from the same class.

Please let us know if you are interested.

**Pull request merged**